### PR TITLE
 refactor: refactor security spec

### PR DESCRIPTION
**Pull request by Ace-Tang**
### Ⅰ. Describe what this PR did

Split #2115   PR into 3 independent PR， here is the first, still need 2 more PRs in the TODO List.

1. add opencontainers/runc/libcontainer/apparmor vendor
2. security parameter include seccomp, apparmor and no-new-privileges
are not implement good, refactor these part when create new spec.

## TODO List
1. use pouchd default spec.
2. remove containerd default spec.

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)

no.

### CODE REVIEW

File: pkg/system/sysinfo.go [hunk 1]
```diff
@@ -3,9 +3,50 @@
...
 import (
+	"io/ioutil"
+	"os"
 	"syscall"
+
+	"golang.org/x/sys/unix"
 )
 
+// SysInfo defines system info on current machine
+type SysInfo struct {
```
Comment (allencloud) [inline pkg/system/sysinfo.go]: In the built-in package, Sysinfo refers to http://docs.studygolang.com/pkg/syscall/#Sysinfo_t : 
```
type Sysinfo_t struct {
    Uptime    int64
    Loads     [3]uint64
    Totalram  uint64
    Freeram   uint64
    Sharedram uint64
    Bufferram uint64
    Totalswap uint64
    Freeswap  uint64
    Procs     uint16
    Pad       uint16
    Pad_cgo_0 [4]byte
    Totalhigh uint64
    Freehigh  uint64
    Unit      uint32
    X_f       [0]byte
    Pad_cgo_1 [4]byte
}
```

So, I am wondering if it is proper to change the definition of `SysInfo`. @Ace-Tang

Comment (Ace-Tang) [reply pkg/system/sysinfo.go]: This is not proper, also I recognize the file name is also not proper, it should be `system.go`, and change the `Sysinfo` to `Info`. That will make sense. @allencloud , WDYT.

Comment (allencloud) [reply pkg/system/sysinfo.go]: Discussed offline. And @Ace-Tang would do some update.

File: vendor/vendor.json [hunk 2]
```diff
@@ -1542,6 +1542,12 @@
 			"revision": "2b8ed96d2a422bf205adcdfa05272dc12c81613b",
 			"revisionTime": "2017-11-08T19:22:26Z"
 		},
+		{
+			"checksumSHA1": "gVVY8k2G3ws+V1czsfxfuRs8log=",
+			"path": "github.com/opencontainers/runc/libcontainer/apparmor",
+			"revision": "ab4a8191679806893207d9c1e507a77ff46dafaa",
+			"revisionTime": "2018-01-12T20:11:04Z"
```
[the remark this answers is not in the document]
Comment (Ace-Tang) [reply vendor/vendor.json]: I think it add by govender, different directory will be a log.

**Comment by Ace-Tang** [comment]
ping @allencloud @fuweid , already updated.

File: Makefile [hunk 3]
```diff
@@ -74,7 +76,7 @@ build: build-daemon build-cli ## build PouchContainer both daemon and cli binari
 build-daemon: modules plugin ## build PouchContainer daemon binary
 	@echo "$@: bin/${DAEMON_BINARY_NAME}"
 	@mkdir -p bin
-	@GOOS=linux go build -ldflags ${DEFAULT_LDFLAGS} -o bin/${DAEMON_BINARY_NAME} -tags 'selinux'
+	@GOOS=linux go build -ldflags ${DEFAULT_LDFLAGS} ${GOBUILD_TAGS}-o bin/${DAEMON_BINARY_NAME}
```
Comment (fuweid) [inline Makefile]: add space before `-o`

I'm just curious about the CI. In my local, make job will fail.

```
$ BUILDTAGS="selinux seccomp apparmor" make
build volume modules
build plugin
build-daemon: bin/pouchd
GOOS=linux go build -ldflags "-X github.com/alibaba/pouch/version.GitCommit=1.0.0-230-gb292535-dirty -X github.com/alibaba/pouch/version.Version="1.0.0" -X github.com/alibaba/pouch/version.ApiVersion="1.24" -X github.com/alibaba/pouch/version.BuildTime=2018-11-01T14:05:11+08:00" -tags "selinux seccomp apparmor"-o bin/pouchd
can't load package: package bin/pouchd: cannot find package "bin/pouchd" in any of:
        /usr/local/go/src/bin/pouchd (from $GOROOT)
        /home/vagrant/go/src/bin/pouchd (from $GOPATH)
Makefile:77: recipe for target 'build-daemon' failed
make: *** [build-daemon] Error 1
```

Comment (fuweid) [reply Makefile]: the CI passed because it doesn't use `make build-daemon`. So please update it @Ace-Tang

**Comment by Ace-Tang** [comment]
```
• Failure [0.966 seconds]
[k8s.io] PodSandbox
/home/travis/gopath/src/github.com/kubernetes-incubator/cri-tools/pkg/framework/framework.go:72
  runtime should support sysctls
  /home/travis/gopath/src/github.com/kubernetes-incubator/cri-tools/pkg/validate/pod.go:84
    should support safe sysctls [It]
    /home/travis/gopath/src/github.com/kubernetes-incubator/cri-tools/pkg/validate/pod.go:95
    Expected
        <string>: 
    to equal
        <string>: 1
    /home/travis/gopath/src/github.com/kubernetes-incubator/cri-tools/pkg/validate/pod.go:253
```

cri tests seems not stable ？

**Comment by fuweid** [comment]
```
• Failure [0.966 seconds]
[k8s.io] PodSandbox
/home/travis/gopath/src/github.com/kubernetes-incubator/cri-tools/pkg/framework/framework.go:72
  runtime should support sysctls
  /home/travis/gopath/src/github.com/kubernetes-incubator/cri-tools/pkg/validate/pod.go:84
    should support safe sysctls [It]
    /home/travis/gopath/src/github.com/kubernetes-incubator/cri-tools/pkg/validate/pod.go:95
    Expected
        <string>: 
    to equal
        <string>: 1
    /home/travis/gopath/src/github.com/kubernetes-incubator/cri-tools/pkg/validate/pod.go:253
cri tests seems not stable ？
```
we will fix in the redesign IO.

**Comment by Starnop** [comment]
@Ace-Tang Please revert it. Something is wrong. In fact, the CI test has failed except the IO problem.

**Comment by allencloud** [comment]
We will discuss this tomorrow. @alibaba/pouch

**Pull request merged**